dbos_transact/dbos.py

```diff
@@ -190,6 +179,144 @@ def get_status(self) -> WorkflowStatus:
 ]
 
 
+class DBOSClassInfo:
+    def __init__(self) -> None:
+        self.def_required_roles: Optional[List[str]] = None
+
+
+class DBOSFuncType(Enum):
+    Unknown = 0
+    Bare = 1
+    Static = 2
+    Class = 3
+    Instance = 4
+
+
+class DBOSFuncInfo:
+    def __init__(self) -> None:
+        self.class_info: Optional[DBOSClassInfo] = None
+        self.func_type: DBOSFuncType = DBOSFuncType.Unknown
+        self.required_roles: Optional[List[str]] = None
+
+
+def get_or_create_class_info(cls: Type[Any]) -> DBOSClassInfo:
```

Comment:
Is there any way these class operations could be in their own file?

Reply:
I still want to straighten out the file structure and the `__index__.py` public API after everything works.  The last thing I want to do before I believe it is reasonably straight is get decorators to work without the DBOS instance.